exonum-java-binding/integration-tests/src/test/java/com/exonum/binding/test/TimeSchemaProxyIntegrationTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.exonum.binding.common.crypto.PublicKey;
+import com.exonum.binding.storage.indices.MapIndex;
+import com.exonum.binding.testkit.EmulatedNode;
+import com.exonum.binding.testkit.EmulatedNodeType;
+import com.exonum.binding.testkit.FakeTimeProvider;
+import com.exonum.binding.testkit.TestKit;
+import com.exonum.binding.testkit.TimeProvider;
+import com.exonum.binding.time.TimeSchema;
+import com.exonum.binding.util.LibraryLoader;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+@RequiresNativeLibrary
+class TimeSchemaProxyIntegrationTest {
```

Comment:
Shan't we also test the case when there is no consolidated time — i.e., before the first blocks?